File: nhglib-core/src/io/github/movementspeed/nhglib/core/entry/NhgEntry.java
package io.github.movementspeed.nhglib.core.entry;

import com.artemis.BaseSystem;
import com.badlogic.gdx.utils.Array;

/**
 * Created by Fausto Napoli on 26/11/2016.
 * Public entry point for games using this library.
 */
public class NhgEntry extends BaseGame {
    public NhgEntry() {
        init(this);
    }

    @Override
    public void onStart() {
        super.onStart();
    }

    @Override
    public void onInitialized() {
        super.onInitialized();
    }

    @Override
    public void onUpdate(float delta) {
        super.onUpdate(delta);
    }

    @Override
    public void onPause() {
        super.onPause();
    }

    @Override
    public void onClose() {
        super.onClose();
    }

    @Override
    public void onResize(int width, int height) {
        super.onResize(width, height);
    }

    @Override
    public void onDispose() {
        super.onDispose();
    }

    @Override
    public Array<BaseSystem> onConfigureEntitySystems() {
        return super.onConfigureEntitySystems();
    }
}

File: nhglib-core/src/io/github/movementspeed/nhglib/core/entry/NhgEntry.kt
package io.github.movementspeed.nhglib.core.entry

import com.artemis.BaseSystem
import com.badlogic.gdx.utils.Array

/**
 * Created by Fausto Napoli on 26/11/2016.
 * Public entry point for games using this library.
 */
open class NhgEntry : BaseGame() {
    init {
        init(this)
    }

    override fun onStart() {
        super.onStart()
    }

    override fun onInitialized() {
        super.onInitialized()
    }

    override fun onUpdate(delta: Float) {
        super.onUpdate(delta)
    }

    override fun onPause() {
        super.onPause()
    }

    override fun onClose() {
        super.onClose()
    }

    override fun onResize(width: Int, height: Int) {
        super.onResize(width, height)
    }

    override fun onDispose() {
        super.onDispose()
    }

    override fun onConfigureEntitySystems(): Array<BaseSystem> {
        return super.onConfigureEntitySystems()
    }
}
